File: backend/internal/entities/post.go
package entities

import (
	"time"
)

type PostId uint

type Post struct {
	Id        PostId    `json:"id"`
	Title     string    `json:"title"`
	Body      string    `json:"body"`
	UserId    UserId    `json:"user_id"`
	UserName  string    `json:"user_name"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

File: frontend/src/shared/models/Post.ts
import { Comment } from '.';

export type PostList = {
  id: number;
  title: string;
  body: string;
  user_id: number;
  username: string;
  created_at?: string;
  updated_at?: string;
};

export type Post = {
  id: number;
  title: string;
  body: string;
  comments: Comment[];
  user_id: number;
  username: string;
  created_at?: string;
  updated_at?: string;
};